localstack/services/cloudformation/engine/template_deployer.py

```diff
@@ -450,33 +448,44 @@ def _resolve_refs_recursively(stack_name, resources, value: dict | list | str |
             item_to_sub[1].update(attr_refs)
 
             for key, val in item_to_sub[1].items():
-                val = resolve_refs_recursively(stack_name, resources, val)
+                val = resolve_refs_recursively(stack_name, resources, mappings, val)
```

Comment:
General observation / nit: With the various refactorings going on right now, it seems a bit brittle to rely mostly on positional args. Technically it should be highlighted by the IDE if incorrect param types are passed to a function, but wondering if we could sprinkle in a few keyword args, to reduce the surface area of incorrect mappings. 🤷

Reply:
Yes! That's something I want to start doing (forcing kwargs usage instead of position arguments).

> Technically it should be highlighted by the IDE if incorrect param types are passed to a function
In practice that hasn't worked out well unfortunately, which is why I'm also of the opinion that kwargs are usually to be preferred where possible.

It's extremely hard to refactor with positional args only especially since most of the GenericBaseModel implementations were using the signatures very inconsistently